osbs/build/spec.py

```diff
@@ -354,8 +357,30 @@ def set_params(self, git_uri=None, git_ref=None,
         self.yum_proxy.value = yum_proxy
         self.koji_parent_build.value = koji_parent_build
 
+        if (signing_intent or compose_ids) and not self.odcs_enabled():
+            raise OsbsValidationException(
+                'signing_intent and compose_ids are allowed only when ODCS is enabled')
+
+        if signing_intent and compose_ids:
+            raise OsbsValidationException(
+                'Please only define signing_intent -OR- compose_ids, not both')
+
+        if compose_ids and yum_repourls:
+            raise OsbsValidationException(
+                'Please only define yum_repourls -OR- compose_ids, not both')
+
+        try:
+            compose_ids and iter(compose_ids)
+        except TypeError:
+            raise OsbsValidationException("compose_ids must be a list")
+
+        self.signing_intent.value = signing_intent
+        self.compose_ids.value = compose_ids or []
         self._populate_image_tag()
 
+    def odcs_enabled(self):
+        return self.odcs_url.value and self.odcs_openidc_secret.value
```

Comment:
Change of requirement: `odcs_openidc_secret` is now optional. This method should only check if `odcs_url` is truthy.

Reply:
ok changed